daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1706,6 +1754,43 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
     funcObj.setOOLAGContext(this)
     funcObj
   }
+
+  def lookupTypeCalculator(typeCalcName:Expression, requiresParse: Boolean, requiresUnparse: Boolean):TypeCalculator[AnyRef, AnyRef]={
+    /*
+     * Every functions that currently uses this defines the type calculator
+     * as their first arguement.
+     * We still take the calculators name as an arguement to avoid relying on this
+     * fact. 
+     */
```

Comment:
Confused by this comment. Why is the arg an Expression called typeCalcName? Why not call it typeCalcExpr ?

Reply:
Also "Every functions" grammar error